File: lib/src/main/java/com/sha/formvalidator/validator/InverseValidator.java
package com.sha.formvalidator.validator;

import android.widget.EditText;

import com.sha.formvalidator.validator.Validator;

/**
 * It's a validator that applies the "NOT" logical operator to the validator it wraps.
 *
 */
public class InverseValidator extends Validator {
    private Validator v;

    public InverseValidator(Validator validator) {
        super(null);
        this.v = validator;
    }

    public InverseValidator(String errorMessage, Validator validator) {
        super(errorMessage);
        this.v = validator;
    }

    public boolean isValid(EditText et) {
        return !v.isValid(et);
    }

}

File: lib/src/main/java/com/sha/formvalidator/validator/InverseValidator.kt
package com.sha.formvalidator.validator

import android.widget.EditText

/**
 * It's a validator that applies the "NOT" logical operator to the validator it wraps.
 *
 */
class InverseValidator : Validator {
    private var v: Validator? = null

    constructor(validator: Validator) : super("") {
        this.v = validator
    }

    constructor(errorMessage: String, validator: Validator) : super(errorMessage) {
        this.v = validator
    }

    override fun isValid(et: EditText): Boolean {
        return !v!!.isValid(et)
    }

}
